### PR TITLE
feat: Current posting and Primary Care Provider Validation 

### PR DESCRIPTION
**Pull request by CIJones11**
## Description

This pull request is the creation of the validation for rule 53 where given a supplied posting is null when the supplied primary care provider is not null. Added unit test for this validation as well.

## Context

This change is required to allowing an exception to be made when both supplied posting and supplied primary care provider is null and no null respectively.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/rules/Breast_Screening_staticRules.json [hunk 1]
```diff
@@ -73,6 +73,10 @@
       {
         "RuleName": "62.ValidateReasonForRemoval.NonFatal",
         "Expression": "!(participant.ReasonForRemoval == \"LDN\" AND  string.IsNullOrEmpty(participant.SupersededByNhsNumber))"
+      },
+      {
+        "RuleName": "53.CurrentPostingAndPrimaryCareProvider.Fatal",
```
Comment (andrewmorris43) [inline application/CohortManager/rules/Breast_Screening_staticRules.json:78]: Just checking, is this definitely a fatal rule?

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 2]
```diff
@@ -992,4 +992,25 @@ private void SetUpRequestBody(string json)
         _request.Setup(r => r.Body).Returns(bodyStream);
     }
 
+    #region Supplied Posting is Null Validation (Rule 53)
+    [TestMethod]
+    [DataRow(null, "PCP123")]
```
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:997]: We don't need to use this DataRow line if there is only one set of data to test against. These can just go directly in the Arrange block

File: application/CohortManager/rules/Breast_Screening_staticRules.json (hunk 1 above)
Comment (CIJones11) [reply application/CohortManager/rules/Breast_Screening_staticRules.json:78]: I am not sure it is actually, that can be changed to NonFatal

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs (hunk 2 above)
Comment (CIJones11) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:997]: In order to make it easier to include extra tests, is it best to just keep it as a DataRow line? What you're saying is true though

File: application/CohortManager/rules/Breast_Screening_staticRules.json (hunk 1 above)
Comment (andrewmorris43) [reply application/CohortManager/rules/Breast_Screening_staticRules.json:78]: Double check with Steph what it should be

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs (hunk 2 above)
Comment (andrewmorris43) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:997]: I'm happy to leave it as is, although I'm not sure any other tests will need to be added for this rule

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 3]
```diff
@@ -992,4 +992,25 @@ private void SetUpRequestBody(string json)
...
+        _participantCsvRecord.Participant.PrimaryCareProvider = primaryCareProvider;
+        var json = JsonSerializer.Serialize(_participantCsvRecord);
+        SetUpRequestBody(json); // Ensure the request body is set up correctly
+
+        // Act
+        var result = await _function.RunAsync(_request.Object);
+
+        // Assert
+        _handleException.Verify(handleException => handleException.CreateValidationExceptionLog(
+            It.Is<IEnumerable<RuleResultTree>>(r => r.Any(x => x.Rule.RuleName == "53.CurrentPostingAndPrimaryCareProvider.Fatal")),
```
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1011]: This should be changed to NonFatal too.

Comment (CIJones11) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs]: Done!

**Review by andrewmorris43 — APPROVED** [review]

**Review by will-larkin-nhs — APPROVED** [review]

**Pull request merged**